### PR TITLE
(2) add assessment section component for teacher dashboard

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link

[Teacher Dashboard page](https://www.notion.so/uwblueprintexecs/Teacher-Dashboard-page-3d48a79f5d9342f5afc7f6d56fe230d5)

## Implementation description
image

- Extract hook for assessment data
- Add assessment section for dashboard page
- Support limit option on backend

## Steps to test
1. View component library page.
2. Check that things visually match the design.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/teacher/dashboard/AssessmentsSection.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import LoadingState from "../../common/info/LoadingState";
+import EmptySessionsMessage from "../../common/info/messages/EmptySessionsMessage";
+import TestSessionListItem from "../view-sessions/TestSessionListItem";
+import useAssessmentDataQuery from "../view-sessions/useAssessmentDataQuery";
+
+const QUERY_DATA_LIMIT_PER_STATUS = 6;
+
+// 3 tabs - this is a quick way of limiting the number of assessments
+// fetched without having complex backend logic
+const QUERY_DATA_LIMIT = QUERY_DATA_LIMIT_PER_STATUS * STATUSES.length;
```
Comment (carissa-tang) [inline frontend/src/components/teacher/dashboard/AssessmentsSection.tsx:27]: with this approach, we could run into the case where we get 18 active assessments which end up as 6, 0, 0 right?

**Review by carissa-tang — APPROVED** [review]
left a question

File: frontend/src/components/teacher/dashboard/AssessmentsSection.tsx (hunk 1 above)
Comment (jfdoming) [reply frontend/src/components/teacher/dashboard/AssessmentsSection.tsx:27]: yeah... my hope was kind of that we could avoid this problem by just not running into it, but you're right I should probably fix it properly by writing a better backend. I'll make a separate pr for this tho

Comment (carissa-tang) [reply frontend/src/components/teacher/dashboard/AssessmentsSection.tsx:27]: i'm thinking we prob will want to sort before limit to provide some guarantee of order

Comment (carissa-tang) [reply frontend/src/components/teacher/dashboard/AssessmentsSection.tsx:27]: but yea separate pr

**Pull request merged**